src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfoExtensions.cs

```diff
@@ -45,7 +51,29 @@ public static ProjectInfo CreateProjectInfo(this ProjectFileInfo projectFileInfo
                 language: LanguageNames.CSharp,
                 filePath: projectFileInfo.FilePath,
                 outputFilePath: projectFileInfo.TargetPath,
-                compilationOptions: projectFileInfo.CreateCompilationOptions());
+                compilationOptions: projectFileInfo.CreateCompilationOptions(),
+                analyzerReferences: analyzerReferences);
+        }
+
+        private static IEnumerable<AnalyzerReference> ResolveAnalyzerReferencesForProject(ProjectFileInfo projectFileInfo, IAnalyzerAssemblyLoader analyzerAssemblyLoader)
+        {
+            return projectFileInfo.Analyzers
+                .GroupBy(x => Path.GetDirectoryName(x))
+                .SelectMany(singleAnalyzerPackageGroup =>
+                {
+                    // Is there better way to figure out entry assembly for specific nuget analyzer package? And is there even entry assembly or is better way to just lookup all referenced assemblies?
+                    return singleAnalyzerPackageGroup
```

Comment:
@mavasani not sure about this. I believe all these analyzer given (I believe it is full path to the dll) must be added to both AddDependencyLocation and created as AanlyzerFileReference? right?

analyzer file reference that doesn't have analyzers in it will be ignored later.

...

so I think the "Where ... xEndsWith.." should be removed. and I dont think hyou need to group things by directory name.

and for file path, having file path in variable name might help to read this code.

..

anyway, I think you can just iterate through analyzers and call AddDependecyLocation and create new AnalyzerFileReference for all those. that will be much simpler and correct.

only thing you might do is make sure Analyzers doesn't have duplicates.